Added CommonJS export.

AWolf81 opened the pull request:
I think it would be good to add CommonJS import option to your service. 

That was the first thing I've noticed after installing it. I'm using Babel with ES6 imports with webpack for bundling my angular app and to have the import work I needed that modification.

Your tests are still passing (not affected by this modification) and I've taken the implementation from `angular-ui-router`.

chrisronline commented:
So sorry for the delay. I'll take a deeper look at this next week. Thanks

chrisronline commented:
So sorry for the super, insanely unreasonable late response, but I have a version ready for testing @AWolf81 if you're still interested?

chrisronline commented:
Check out `v0.1.0`

AWolf81 commented:
OK, no problem and thanks for adding CommonJS/AMD support. I'm still interested.

I've added v0.1.0 to my project and removed my fork but it's not working. I think it's a problem with the window object in loader function. (Webpack is using AMD to load the module.)

I'm getting the following error message in browser console:
`Error: Cannot find module '[object Window]'.`

With the following modifications it is working for me:

**Removed window object from module loader function**

    (function (root, factory) {
      if(typeof define === 'function' && define.amd) {
        define(['angular'], factory);
      } else if(typeof module === 'object' && module.exports) {
          module.exports = factory(require('angular'));
      } else {
          root['angular-promise-cache'] = factory(root['angular']);
      }
    }(this, function(angular) {

**To use localstorage I've added it like this with $window dependency injection:**

    angular.module('angular-promise-cache', [])
        .factory('promiseCache', ['$q', '$rootScope', '$window', function($q, $rootScope, $window) {

**And load localStorage object:**

    try {
        ls = $window.localStorage;
      } catch (e) {

**Finally I'm returning the module name (inside the closure)**
    
    return 'angular-promise-cache';

chrisronline commented:
What's your setup like? Is it a public github repo? I'd like to iterate/test to make sure it works in all ways before pushing the next fix. Thanks!

chrisronline commented:
Okay thanks - I'll look at this today

chrisronline commented:
@AWolf81 Try using `v0.1.1`

chrisronline commented:
I've been slacking on this - is this still not working for you?

AWolf81 commented:
OK, no problem. I haven't worked on Angular 1 for a while and I think I'm moving to Angular 2 soon. But I'm still learning ng2.

I've created an example app with ES6 & Webpack to test your new version. Please have a look [here](https://github.com/AWolf81/angular-promise-cache-es6-example).

I think we can close this PR because importing your service is working now.

chrisronline commented:
Sounds great. Thanks!
